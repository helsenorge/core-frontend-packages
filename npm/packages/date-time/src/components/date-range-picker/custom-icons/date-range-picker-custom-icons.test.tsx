import * as React from 'react';
import { mount } from 'enzyme';
import Icon from '@helsenorge/designsystem-react/components/Icon';
import ChevronLeft from '@helsenorge/designsystem-react/components/Icons/ChevronLeft';
import ChevronRight from '@helsenorge/designsystem-react/components/Icons/ChevronRight';
import ArrowIcon from './arrow-icon';
import InputIcon from './input-icon';
import HeaderNavIcon from './header-nav-icon';

describe('DateRangePicker icons', () => {
  describe('Gitt at ArrowIcon skal rendres', () => {
    describe('Når det brukes optional Prop className', () => {
      const wrapper = mount(<ArrowIcon className={'custom-classname-1'} />);
      it('Så rendres det riktig ikon med custom className', () => {
        expect(wrapper.render()).toMatchSnapshot();
        expect(wrapper.find('svg.custom-classname-1').length).toBe(1);
      });
    });
  });

  describe('Gitt at InputIcon skal rendres', () => {
    const onClickMock = jest.fn();
    const wrapper = mount(<InputIcon onClick={onClickMock} className={'custom-classname-2'} />);
    describe('Når det brukes optional Prop className', () => {
      it('Så rendres det riktig ikon med custom className', () => {
        expect(wrapper.render()).toMatchSnapshot();
        expect(wrapper.find('svg.custom-classname-2').length).toBe(1);
      });
    });
    describe('Når det klikkes på ikonet', () => {
      wrapper.find('span').simulate('click');
      wrapper.update();
      it('Så rendres det riktig ikon med custom className', () => {
        expect(onClickMock).toHaveBeenCalled();
      });
    });
  });

  describe('Gitt at HeaderNavIcon skal rendres', () => {
    describe('Når det brukes direction prev', () => {
      const wrapper = mount(<HeaderNavIcon direction={'prev'} className={'custom-classname-3'} />);
      it('Så rendres det riktig svgIcon og riktig className', () => {
        expect(wrapper.find('span').prop('className')).toEqual('DayPickerNavigation_leftButton__horizontalDefault custom-classname-3');
        expect(wrapper.find(Icon).prop('svgIcon')).toBe(ChevronLeft);
        expect(wrapper.render()).toMatchSnapshot();
      });
    });
    describe('Når det brukes direction next', () => {
      const wrapper = mount(<HeaderNavIcon direction={'next'} className={'custom-classname-4'} />);
      it('Så rendres det riktig svgIcon og riktig className', () => {
        expect(wrapper.find('span').prop('className')).toEqual('DayPickerNavigation_rightButton__horizontalDefault custom-classname-4');
        expect(wrapper.find(Icon).prop('svgIcon')).toBe(ChevronRight);
        expect(wrapper.render()).toMatchSnapshot();
      });
    });
  });
});
