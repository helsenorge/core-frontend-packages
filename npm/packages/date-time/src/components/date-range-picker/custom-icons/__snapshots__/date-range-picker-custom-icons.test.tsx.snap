// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangePicker icons Gitt at ArrowIcon skal rendres Når det brukes optional Prop className Så rendres det riktig ikon med custom className 1`] = `
<span>
  <svg
    aria-hidden="true"
    class="hnds-style-icon custom-classname-1"
    data-analyticsid="icon"
    fill="black"
    height="36"
    role="img"
    style="min-width: 36px; min-height: 36px;"
    viewBox="0 0 48 48"
    width="36"
  >
    <path
      d="M27.075 5.929l-1.142 1.377 13.044 15.721H5.929v1.948h33.048L25.933 40.692l1.142 1.379 14.996-18.07z"
    />
  </svg>
</span>
`;

exports[`DateRangePicker icons Gitt at HeaderNavIcon skal rendres Når det brukes direction next Så rendres det riktig svgIcon og riktig className 1`] = `
<span
  class="DayPickerNavigation_rightButton__horizontalDefault custom-classname-4"
  tabindex="0"
>
  <svg
    aria-hidden="true"
    class="hnds-style-icon "
    data-analyticsid="icon"
    fill="#126F87"
    height="36"
    role="img"
    style="min-width: 36px; min-height: 36px;"
    viewBox="0 0 48 48"
    width="36"
  >
    <path
      d="M19.171 32.951l1.409 1.42L31.031 24 20.58 13.63l-1.409 1.42 9.02 8.95z"
    />
  </svg>
</span>
`;

exports[`DateRangePicker icons Gitt at HeaderNavIcon skal rendres Når det brukes direction prev Så rendres det riktig svgIcon og riktig className 1`] = `
<span
  class="DayPickerNavigation_leftButton__horizontalDefault custom-classname-3"
  tabindex="0"
>
  <svg
    aria-hidden="true"
    class="hnds-style-icon "
    data-analyticsid="icon"
    fill="#126F87"
    height="36"
    role="img"
    style="min-width: 36px; min-height: 36px;"
    viewBox="0 0 48 48"
    width="36"
  >
    <path
      d="M30.66 32.951l-1.409 1.42L18.8 24l10.451-10.37 1.409 1.42L21.64 24z"
    />
  </svg>
</span>
`;

exports[`DateRangePicker icons Gitt at InputIcon skal rendres Når det brukes optional Prop className Så rendres det riktig ikon med custom className 1`] = `
<span>
  <svg
    aria-hidden="true"
    class="hnds-style-icon custom-classname-2"
    data-analyticsid="icon"
    fill="black"
    height="36"
    role="img"
    style="min-width: 36px; min-height: 36px;"
    viewBox="0 0 48 48"
    width="36"
  >
    <path
      d="M32.86 22.986a.984.984 0 010 1.967.984.984 0 010-1.967zm0 3.03a2.049 2.049 0 002.046-2.047 2.049 2.049 0 00-2.046-2.046 2.049 2.049 0 00-2.046 2.046c0 1.13.918 2.047 2.046 2.047zm6.21-7.864H8.68v-6.194h3.846v1.161a2.243 2.243 0 002.241 2.241 2.243 2.243 0 002.241-2.24v-1.162H30.62v1.161a2.244 2.244 0 002.241 2.241 2.243 2.243 0 002.241-2.24v-1.162h3.97v6.194zM8.678 35.982H39.07v-16.53H8.678v16.53zm5.148-26.273c0-.518.422-.94.941-.94.52 0 .941.422.941.94v3.41a.942.942 0 01-1.882 0V9.71zm18.093 0c0-.518.422-.94.941-.94.52 0 .941.422.941.94v3.41a.942.942 0 01-1.882 0V9.71zm3.182.95v-.95a2.243 2.243 0 00-2.24-2.24 2.244 2.244 0 00-2.242 2.24v.95H17.01v-.95a2.243 2.243 0 00-2.242-2.24 2.243 2.243 0 00-2.24 2.24v.95H7.378V37.28h32.992V10.658h-5.27zm-8.157 16.167a.984.984 0 110 1.969.984.984 0 010-1.969zm0 3.03a2.049 2.049 0 002.047-2.047 2.049 2.049 0 00-2.047-2.046 2.049 2.049 0 00-2.046 2.046c0 1.13.918 2.047 2.046 2.047z"
      fill-rule="evenodd"
    />
  </svg>
</span>
`;
